lib/utils/config.ts

```diff
@@ -165,3 +176,48 @@ export async function validateDevOptionAndWarn(
 		and exposes network ports such as 2375 that allows unencrypted access to balenaEngine.
 		Therefore, development mode should only be used in private, trusted local networks.`);
 }
+
+/**
+ * Chech whether the `--secureBoot` option of commands related to OS configuration
+ * such as `os configure` and `config generate` is compatible with a given
+ * device type and balenaOS version, and print a warning regarding the
+ * consequences of using that option.
+ */
+export async function validateSecureBootOptionAndWarn(
+	secureBoot?: boolean,
+	deviceManifest?: any,
+	version?: string,
+	logger?: import('./logger'),
+) {
+	if (!secureBoot) {
+		return;
+	}
+
+	const deviceManifestObj: { [index: string]: any } = deviceManifest;
+	const deviceType = deviceManifestObj['yocto']['machine'];
+
+	const { ExpectedError } = await import('../errors');
+	const supportedDeviceTypes = ['generic-amd64'];
```

Comment:
Ideally this should come from the DT contract (or the device-type.json :disappointed: ), so that we don't need to release new versions of the CLI every time we add support for secure boot to an extra DT.

For now, let's just move this const definition outside of the function at the top level to raise its importance vs other vars.

Reply:
hey @otaviojacobi I agree - the correct place is the future equivalent to the `device-type.json` as there are two sides:

* The device type contract specified hw capabilities - whether the hardware has the secure boot capability
* The OS contract should state that the software supports it.

So the right way to do it would be for the CLI to query the OS release for its contract and use that.

The options were:

1) Hardcoding the list - this would allow us to merge this quickly and continue with the SB release
2) Extend `device-type.json` - this would be quick but we must not use this file any more
3) Extend the OS contract to add the secure boot setting, and have the CLI use it. For me this is the right thing, but it will take longer and will delay the SB feature.

So I will do as you say and move it outside the function, merge it, and work on 3) so we don't have to update the CLI in the future.

Reply:
Yea I agree. Having
* the DT contract on whether the HW can support secure boot (so that automation can use it)
* AND a field in the hostApp release contract is probably the best option, as the clearest way to communicate whether a given OS release supports.

Even though it would make it harder to find the min OS version that added support for secure boot, I don't think it matters that much. A generic error like "Secure boot is not supported by this OS Version" sgtm.

Reply:
actually @thgreasi I think I going to dedicate a few hours to this approach as it's the right one and simplifies things - no need to check for DT or version compatibility, if the OS contract says it's supported that's it.

Reply:
The code now uses the OS release contract